### ISSUE TITLE
provider: deploys stale deploments upon restart

### ISSUE THREAD

**Issue by gosuri** [ovrclk/akash#380]
seems like a CRD bug, more in the video: https://www.dropbox.com/s/5t3heu30z9biqau/provider-crd.mov?dl=0

**Comment by gosuri** [comment]
re-opening due to return of unexpected behavior:

## `akash query deployment -k gosuri:`

```
Deployment(s)
=============

Deployment ID                                                   	Tenant ID                               	State 	Version                                                         
-------------                                                   	---------                               	----- 	-------                                                         
                                                                	                                        	      	                                                                
042b1ba2c0bdb52437528c4f5a28bc5d278bfc76ea369dc5ff8445baf9dcb6ff	981d5831203c281969a609be952b5b2456f775b2	CLOSED	230341e751159c7f597ebbcc592a531a81233e70bc0f27f347068dd20289be3c
27b32f7b6c0702a22b525517f49dc366fcb5edf02faf4a04e636281ea2aacea5	981d5831203c281969a609be952b5b2456f775b2	CLOSED	f31f38b28fd7abf9078c0452b2b4a7af0a519e3a352aa6408c44e9633441ebf2
59abd24c671dd40d67b8dd979c179cc94c1e461e6e45248ab9eab84b8ccc5310	981d5831203c281969a609be952b5b2456f775b2	CLOSED	230341e751159c7f597ebbcc592a531a81233e70bc0f27f347068dd20289be3c
5e988ed5c9c4ed00b1f7f722c538e1313735f4a7905caf5bf9466f8499200049	981d5831203c281969a609be952b5b2456f775b2	CLOSED	230341e751159c7f597ebbcc592a531a81233e70bc0f27f347068dd20289be3c
8e0e627d85c2487c685aca943df217b8f5605d97a2c9fcae4d17978be467ece7	981d5831203c281969a609be952b5b2456f775b2	CLOSED	230341e751159c7f597ebbcc592a531a81233e70bc0f27f347068dd20289be3c
e0e035467bba668eb47f027fc2a4da55f1078174c4009b8a48bd57992dc741d0	981d5831203c281969a609be952b5b2456f775b2	CLOSED	fbcdb5807806eb64f7f7e1147ef4c7d28e47c1956355647c21bb66ae3f3ddf96
f6f1e9762e687c511df8e120d2d6e82ad3ca0e0ed4ddfdda6c136bb7fd6d5a43	981d5831203c281969a609be952b5b2456f775b2	CLOSED	d375b0e1abe96db4bc278b20059b9d6bf34ed0540abb2e5b4d535291a8f3cfee
fca47daee9a0948525a2db2384bdf17bc218d437a5f6e433920b99b1fe5dcd17	981d5831203c281969a609be952b5b2456f775b2	CLOSED	230341e751159c7f597ebbcc592a531a81233e70bc0f27f347068dd20289be3c
```

## `kubectl get manifests.akash.network -A`:
```
NAMESPACE   NAME                                       AGE
lease       a2715ba684c5cf69d72b8a923cac094af2d8605f   50m
lease       7c5c60888a793602b1d4f600402d321b71e4ccbf   44m
lease       e1edec22bd999e76835fb3486f8ffb2d33625515   41m
lease       c767a9ace0b719b7a520aa39f8099e43cdb9d3c0   37m
```

##  `kubectl get pods -A`:

```
NAMESPACE     NAME                                                            READY   STATUS      RESTARTS   AGE
kube-system   coredns-66f496764-fckkf                                         1/1     Running     0          16h
kube-system   helm-install-traefik-7x4qb                                      0/1     Completed   0          16h
kube-system   tiller-deploy-767d9b9584-gwfjt                                  1/1     Running     0          16h
kube-system   svclb-traefik-xt5cf                                             3/3     Running     0          16h
kube-system   traefik-d869575c8-pjpsj                                         1/1     Running     0          16h
default       kernel-metrics-prometheus-node-exporter-ffxkv                   1/1     Running     0          15h
default       kernel-metrics-prometheus-kube-state-metrics-654b569c9f-jn67q   1/1     Running     0          15h
default       kernel-metrics-prometheus-pushgateway-67dd54ff9f-k47zn          1/1     Running     0          15h
default       kernel-status-grafana-7868c66445-v6m8b                          1/1     Running     0          15h
default       kernel-metrics-server-578b6d8ff6-w278m                          1/1     Running     0          15h
default       kernel-metrics-prometheus-alertmanager-56cfc69859-bj4kk         2/2     Running     0          15h
default       kernel-metrics-prometheus-server-59797cc6b7-m5gkp               2/2     Running     0          15h
default       node2-akash-node-69fb7b6645-ktj9b                               1/1     Running     0          15h
default       node3-akash-node-566f6fd485-j42rq                               1/1     Running     0          15h
default       node1-akash-node-6fcdcbddbf-4pfv5                               1/1     Running     2          15h
kube-system   csi-node-4zvg6                                                  2/2     Running     0          8h
kube-system   csi-packet-controller-0                                         3/3     Running     0          8h
default       locke-akash-provider-55c97b7479-lflvk                           1/1     Running     0          10m
```

## redeploy provider

```
helm del --purge locke
helm install akash-provider --name "locke" --set "ingress.domain=locke.akashtest.net"
```

```
$ kubectl get pods -A

NAMESPACE                                  NAME                                                            READY   STATUS      RESTARTS   AGE
kube-system                                coredns-66f496764-fckkf                                         1/1     Running     0          16h
kube-system                                helm-install-traefik-7x4qb                                      0/1     Completed   0          16h
kube-system                                tiller-deploy-767d9b9584-gwfjt                                  1/1     Running     0          16h
kube-system                                svclb-traefik-xt5cf                                             3/3     Running     0          16h
kube-system                                traefik-d869575c8-pjpsj                                         1/1     Running     0          16h
default                                    kernel-metrics-prometheus-node-exporter-ffxkv                   1/1     Running     0          15h
default                                    kernel-metrics-prometheus-kube-state-metrics-654b569c9f-jn67q   1/1     Running     0          15h
default                                    kernel-metrics-prometheus-pushgateway-67dd54ff9f-k47zn          1/1     Running     0          15h
default                                    kernel-status-grafana-7868c66445-v6m8b                          1/1     Running     0          15h
default                                    kernel-metrics-server-578b6d8ff6-w278m                          1/1     Running     0          15h
default                                    kernel-metrics-prometheus-alertmanager-56cfc69859-bj4kk         2/2     Running     0          15h
default                                    kernel-metrics-prometheus-server-59797cc6b7-m5gkp               2/2     Running     0          15h
default                                    node2-akash-node-69fb7b6645-ktj9b                               1/1     Running     0          15h
default                                    node3-akash-node-566f6fd485-j42rq                               1/1     Running     0          15h
default                                    node1-akash-node-6fcdcbddbf-4pfv5                               1/1     Running     2          15h
kube-system                                csi-node-4zvg6                                                  2/2     Running     0          8h
kube-system                                csi-packet-controller-0                                         3/3     Running     0          8h
default                                    locke-akash-provider-55c97b7479-xvf6n                           1/1     Running     0          17s
a2715ba684c5cf69d72b8a923cac094af2d8605f   web-cd56977cb-mpxjz                                             1/1     Running     0          16s
a2715ba684c5cf69d72b8a923cac094af2d8605f   web-cd56977cb-9l5d2                                             1/1     Running     0          16s
c767a9ace0b719b7a520aa39f8099e43cdb9d3c0   web-5548b476d4-mbk9p                                            1/1     Running     0          16s
7c5c60888a793602b1d4f600402d321b71e4ccbf   web-cd56977cb-66ncf                                             1/1     Running     0          16s
e1edec22bd999e76835fb3486f8ffb2d33625515   web-5548b476d4-s2sx7                                            1/1     Running     0          16s
c767a9ace0b719b7a520aa39f8099e43cdb9d3c0   web-5548b476d4-xrwxl                                            1/1     Running     0          16s
7c5c60888a793602b1d4f600402d321b71e4ccbf   web-cd56977cb-s56sk                                             1/1     Running     0          16s
e1edec22bd999e76835fb3486f8ffb2d33625515   web-5548b476d4-r9k6m                                            1/1     Running     0          16s

```

**Comment by gosuri** [comment]
a potential solution would be to remove stale manifests when upon deployment termination by tenant:

```
$ kubectl delete ns a2715ba684c5cf69d72b8a923cac094af2d8605f c767a9ace0b719b7a520aa39f8099e43cdb9d3c0 7c5c60888a793602b1d4f600402d321b71e4ccbf

kubectl get manifests -A | grep lease | awk '{print $2}' | xargs kubectl delete manifest -n lease
```

## `kubectl get pods -A`:
```
NAMESPACE                                  NAME                                                            READY   STATUS        RESTARTS   AGE
kube-system                                coredns-66f496764-fckkf                                         1/1     Running       0          16h
kube-system                                helm-install-traefik-7x4qb                                      0/1     Completed     0          16h
kube-system                                tiller-deploy-767d9b9584-gwfjt                                  1/1     Running       0          16h
kube-system                                svclb-traefik-xt5cf                                             3/3     Running       0          16h
kube-system                                traefik-d869575c8-pjpsj                                         1/1     Running       0          16h
default                                    kernel-metrics-prometheus-node-exporter-ffxkv                   1/1     Running       0          16h
default                                    kernel-metrics-prometheus-kube-state-metrics-654b569c9f-jn67q   1/1     Running       0          16h
default                                    kernel-metrics-prometheus-pushgateway-67dd54ff9f-k47zn          1/1     Running       0          16h
default                                    kernel-status-grafana-7868c66445-v6m8b                          1/1     Running       0          16h
default                                    kernel-metrics-server-578b6d8ff6-w278m                          1/1     Running       0          16h
default                                    kernel-metrics-prometheus-alertmanager-56cfc69859-bj4kk         2/2     Running       0          16h
default                                    kernel-metrics-prometheus-server-59797cc6b7-m5gkp               2/2     Running       0          16h
default                                    node2-akash-node-69fb7b6645-ktj9b                               1/1     Running       0          15h
default                                    node3-akash-node-566f6fd485-j42rq                               1/1     Running       0          15h
default                                    node1-akash-node-6fcdcbddbf-4pfv5                               1/1     Running       2          15h
kube-system                                csi-node-4zvg6                                                  2/2     Running       0          8h
kube-system                                csi-packet-controller-0                                         3/3     Running       0          8h
default                                    locke-akash-provider-55c97b7479-fdrgf                           1/1     Running       0          66s
```

**Comment by boz** [comment]
hmm let's see if the fix for #404

**Comment by gosuri** [comment]
Great. Testing on testnet
